fakeredis/commands_mixins/json_mixin.py

```diff
@@ -0,0 +1,605 @@
+            # versions of the module. For details, see:
+            # https://github.com/RedisJSON/RedisJSON/issues/145
+            pass
+
+        path_count = len(args)
+        cached_value = json.loads(name.value)
+
+        callers = starmap(
+            methodcaller,
+            zip(
+                repeat("find", path_count),
+                repeat(cached_value, path_count),
+            ),
+        )
+        path_values = tuple(
+            map(
+                attrgetter("value"),
+                chain.from_iterable(
+                    (
+                        caller(path)
+                        for caller, path in zip(
+                            callers,
+                            args,
+                        )
+                    )
+                ),
+            )
+        )
+
+        if len(args) == 1:
+            path_values = path_values[0]
+
+        return json.dumps(path_values).encode()
+
+    @command(
+        name="JSON.SET",
+        fixed=(Key(), bytes, JSONObject),
+        repeat=(bytes,),
+    )
+    def json_set(
+        self: FakeSocket,
+        name: CommandItem,
+        path: bytes,
+        obj: JsonType,
+        flag: Optional[Literal[b"NX", b"XX"]] = None,
+    ) -> Any:
+        """Set the JSON value at key `name` under the `path` to `obj`.
+
+        if `flag` is b"NX", set `value` only if it does not exist.
+        if `flag` is b"XX", set `value` only if it exists.
+
+        For the purpose of using this within a pipeline, this command is also
+        aliased to JSON.SET.
+
+        For more information see `JSON.SET <https://redis.io/commands/json.set>`_.
+        """
+        cached_value, path = (
+            json.loads(name.value or b"null"),
+            parse_jsonpath(format_jsonpath(path)),
+        )
+
+        setter = partial(path.update_or_create, cached_value)
+
+        if flag and flag not in (b"NX", b"XX"):
+            raise helpers.SimpleError(f"Unknown or unsupported `JSON.SET` flag: {flag}")
+        elif (flag == b"NX" and path.find(cached_value)) or (flag == b"XX" and not path.find(cached_value)):
+            setter = partial(null_op, cached_value)
+
+        cached_value, name.value = (
+            name.value,
+            json.dumps(setter(obj)).encode(),
+        )
+
+        name.writeback()
+
+        return helpers.OK if cached_value != name.value else None
+
+    @command(
+        name="JSON.MGET",
+        fixed=(Key(), bytes),
+        repeat=(bytes,),
+    )
+    def json_mget(
+        self: FakeSocket,
+        keys: list[CommandItem],  # This seems like the wrong type 🤔
+        path: bytes,
+    ) -> list[JsonType]:
+        """Get the objects stored as a JSON values under `path`. `keys` is a
+        list of one or more keys.
+
+        For more information see `JSON.MGET
+        <https://redis.io/commands/json.mget>`_.
+        """
+        raise NotImplementedError
+
+    @command(
```

Comment:
If the method exists and is not implemented, the `supported.py` script will say it is.
Remove unimplemented methods entirely.

Reply:
Applied